build.gradle

```diff
@@ -221,6 +223,8 @@ project(':iceberg-core') {
       exclude group: 'org.slf4j', module: 'slf4j-log4j12'
     }
 
+    implementation 'org.apache.httpcomponents.client5:httpclient5'
```

Comment:
This is already included with Spark, right? So we should make sure that it is excluded from the Spark runtime Jars.

Reply:
I don't think this is included with Spark. Let me double check.

Reply:
`spark_hive` has an older version, which comes with a different module name. `org.apache.httpcomponents:httpclient:4.5.13` for Spark 3.2.1: https://mvnrepository.com/artifact/org.apache.spark/spark-hive_2.12/3.2.1

We might need to exclude some packages, but I think they've mostly all been renamed as of the name change to include `5` in the artifact name and the org.

Reply:
How large is this dependency if we add it? We might just want to add and shade it everywhere if it isn't simple to rely on the one already in the classpath.

Reply:
The JAR itself is about 779 kb (so not small)

![image](https://user-images.githubusercontent.com/9833362/155034136-3695a704-4863-4988-8354-506fa7d076c1.png)

And then it has the following compile time dependencies as well

![image](https://user-images.githubusercontent.com/9833362/155034203-220531a3-6c8d-4961-abac-e14cee20b010.png)